File: src/python/som_segmentation.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Tudor Berariu, 2014

from PIL import Image
import copy
import sys
from learning_rate import *
from radius import *
from neighbourhood import *

def som_segmentation(orig_file_name, n):
    ## După rezolvarea Exercițiilor 2, 3 și 4
    ## în fișierele learning_rate.py, radius.py și neighbourhood.py
    ## rezolvați aici Exercițiile 5 și 6

    orig_img = Image.open(orig_file_name)
    orig_pixels = list(orig_img.getdata())
    orig_pixels = [(o[0]/255.0, o[1]/255.0, o[2]/255.0) for o in orig_pixels]


    ## Exercițiul 5: antrenarea rețelei Kohonen
    ## Exercițiul 5: completați aici:

    ## Exercițiul 5: ----------

    ## Exercițiul 6: crearea imaginii segmentate pe baza ponderilor W
    ## Exercițiul 6: porniți de la codul din show_neg
    ## Exercițiul 6: completați aici:

    ## Exercițiul 6: ----------
    pass

if __name__ == "__main__":
    som_segmentation(sys.argv[1], sys.argv[2])

File: src/octave/som_segmentation.m
function som_segmentation( orig_file_name, n )
    %SOM_SEGMENTATION Segmentează o imaginie utilizand SOM

    orig_img = imread(orig_file_name);
    orig_pixels = cast(orig_img,'double');
    orig_pixels = orig_pixels/255.0;

    %% După rezolvarea Taskurilor 2, 3 și 4
    %% în fișierele learning_rate.m, radius.m și neighbourhood.m,
    %% rezolvați aici Taskurile 5 și 6.


    %% Exercițiul 5: antrenarea retelei Kohonen
    %% Exercițiul 5: completați aici
    W = rand(n,n,3) / 2;
    %% Exercițiul 5: ----------

    %% Exercițiul 6: compunerea imaginii segmentate pe baza ponderilor W
    %% Exercițiul 6: porniți de la codul din negative.m
    %% Exercițiul 6: completați aici

    % seg_pixels = TODO
    % seg_img = cast(seg_pixels,'uint8');
    % seg_file_name = TODO
    % image(seg_img)
    % imwrite(seg_img, seg_file_name);

    %% Exercițiul 6: ----------
end

%% Tudor Berariu, 2014